index.js

```diff
@@ -45,26 +47,42 @@ var ERROR_TYPES = {
  * This library stores no default configuration of its own. It relies on the application/service
  * it is embedded in to supply this information.
  *
- * @param config
+ * @param appUA an identifying user agent string for your app, compatible with user agent formatting
+ * as per https://tools.ietf.org/html/rfc7231#section-5.5.3 e.g. 'my-app', 'my-app/0.1',
+ * 'my-app/0.1 (Ubuntu/12.04; nodejs/0.10.13)`
+ * @param config object containing config
  * @constructor
  */
-var PersonaClient = function (config) {
-    this.config = config || {};
-
-    var requiredAttributes = [
-        'persona_host', 'persona_port', 'persona_scheme',
-        'persona_oauth_route'
-    ];
-
-    for (var i = 0; i < requiredAttributes.length; i++) {
-        var attribute = requiredAttributes[i];
+var PersonaClient = function (appUA, config) {
```

Comment:
In the persona-php-client, the variable is passed in via config (it's still mandatory) - are we worried that they are different? It's also called "userAgent" too.